interpreter/interpreter.go

```diff
@@ -177,6 +178,31 @@ func (itrp *Interpreter) doStatement(ctx context.Context, stmt semantic.Statemen
 	return nil, nil
 }
 
+// If the option is "now", evaluate the function and store in the execution
+// dependencies.
+func (irtp *Interpreter) evaluateNowOption(ctx context.Context, name string, init values.Value) {
+	if name != "now" {
+		return
+	}
+	if !execdeps.HaveExecutionDependencies(ctx) {
+		return
+	}
+
+	// Evaluate now.
+	nowTime, err := init.Function().Call(ctx, nil)
+	if err != nil {
+		return
+	}
+	now := nowTime.Time().Time()
```

Comment:
The implementation of `now()` does not use the `ctx` variable. And it doesn't look like it has been updated which means that unfortunately `nowTime` will always be the current time of execution. I.e. it will never be overridden.

Reply:
https://github.com/influxdata/flux/pull/2773 is needed in order for tasks to override the default `now` time outside of flux

Reply:
Ah, I didn't realize there was also a now() function that needs to be updated.